sigstore-java/src/main/java/dev/sigstore/tuf/Updater.java

```diff
@@ -104,20 +105,24 @@ public void updateMeta() throws IOException, NoSuchAlgorithmException, InvalidKe
     updateRoot();
     var oldTimestamp = trustedMetaStore.findTimestamp();
     updateTimestamp();
-    if (Objects.equals(oldTimestamp.orElse(null), trustedMetaStore.getTimestamp())
-        && trustedMetaStore.findSnapshot().isPresent()
-        && trustedMetaStore.findTargets().isPresent()) {
-      return;
-    }
-    // if we need to update or we can't find targets/timestamps locally then grab new snapshot and
-    // targets from remote
-    updateSnapshot();
-    updateTargets();
+    if (!Objects.equals(oldTimestamp.orElse(null), trustedMetaStore.getTimestamp())
+        || trustedMetaStore.findSnapshot().isEmpty()
+        || trustedMetaStore.findTargets().isEmpty()) {
+      // if we need to update or we can't find targets/snapshots locally then grab new snapshot and
+      // targets from remote
+      updateSnapshot();
+      updateTargets();
+    }
+    metaUpdated = true;
```

Comment:
I'm a bit confused by what this is tracking?  why would it be class level?

Reply:
Yeah this part is strange, but basically a call to "downloadTargets" should call an update if within the context of this updater an update has never occurred. (tuf conformance expects it. Also the go/python clients work this way)

Reply:
I don't remember, but I don't think I designed the Updater to be necessarily single-shot. Even if it was this is kind of ugly for me, but it's difficult for me to understand what the tuf conformance has to do with it.

Reply:
it wasn't designed to be single shot. I can instead move this logic to the conformance wrapper. Lemme do that. We can revisit this later.